core/src/verified_vote_packets.rs

```diff
@@ -229,22 +245,35 @@ impl VerifiedVotePackets {
                     }
                     let slot = vote.last_voted_slot().unwrap();
                     let hash = vote.hash();
+                    let timestamp = vote.timestamp();
 
                     match (vote, is_full_tower_vote_enabled) {
                         (VoteStateUpdate(_), true) => {
-                            let latest_gossip_slot = match self.0.get(&vote_account_key) {
-                                Some(vote) => vote.get_latest_gossip_slot(),
-                                _ => 0,
-                            };
+                            let (latest_gossip_slot, latest_gossip_hash, latest_timestamp) =
+                                match self.0.get(&vote_account_key) {
+                                    Some(vote) => (
+                                        vote.get_latest_gossip_slot(),
+                                        vote.get_latest_gossip_hash(),
+                                        vote.get_latest_gossip_timestamp(),
+                                    ),
+                                    _ => (0, Hash::default(), None),
+                                };
                             // Since votes are not incremental, we keep only the latest vote
-                            if slot > latest_gossip_slot {
+                            // If the vote is for the same slot we will only allow it if
+                            //  1) It is for a different hash (voting for duplicate blocks)
```

Comment:
Added hash as well here because the previous impl used `(Slot, Hash)` as the key

Reply:
discussed offline: hash is not relevant here, only using ts for tiebreaker